components/dashboard/DashboardAssignments.js

```diff
@@ -23,27 +23,59 @@ class DashboardAssignments extends Component {
     const assignmentFilters = [
       { name: 'Teams', id: 'teams' },
       { name: 'Favorites', id: 'favorites' },
-      { name: 'All', id: 'all' }
+      { name: 'All', id: 'userContributions' }
     ]
-
     let teamAssignments = sortBy(prop('team_priority'), assignments).map(task => {
       return {
         priority: task.team_priority ? `team priority: ${task.team_priority}` : task.priority,
         name: task.name,
         assigned_by: task.team_name,
-        campaign_hashtag: task.campaign_hashtag
+        campaign_hashtag: task.campaign_hashtag,
+        source: 'ASSIGNMENT'
+      }
+    })
+    let userFavorites = favorites.map(favorite => {
```

Comment:
@alyssadelaine @kamicut more of a question on object mapping; it seems like I shouldn't have to list out every key value pair for each object just to add a new pair, 'source.' I'm doing this both here in 'userFavorites' and in 'userContributions.' Is there a simpler or better way to do this?

Reply:
@LanesGood You're not listing out the key:value pair; you're filtering the original array to create _an entirely new one_.

The map function says, "Take this array and change it in the following way: [...]" so that's what you're doing for each of these arrays before saving the selected items to the new one. 

Your code doesn't know that this is the group of team assignments until you tell it that, in the same way that it doesn't know the source until you define it.

Another way to do this might be to add each of the array items to the corresponding new array within the map function, but @kamicut will have to correct me if that's not very JS.